Crawler/Analyzers/UtilAnalyzers/PunctuationAnalyzer.cs

```diff
@@ -80,5 +91,45 @@ private List<int> WordsCountBetweenPunctuations(List<Token> tokens)
 
             return wordsCount;
         }
+
+        public double CalculatePassiveVoiceSentencesPercentage(List<PosTagToken> posTagTokens)
+        {
+	        var toBeForms = ExtractWordsFromFile(config.Value.ToBeFormsFile);
+	        var passiveVoiceSentencesCount = 0;
+	        var indexOfNextSentenceEnd = posTagTokens.FindIndex(token => token.ExtendedType == ".");
+            var totalSentencesCount = posTagTokens.Count(token => token.ExtendedType == ".");
+
+	        if (totalSentencesCount == 0)
+	        {
+		        totalSentencesCount = 1;
+	        }
+
+	        while (indexOfNextSentenceEnd != -1)
+	        {
+		        var sentence = posTagTokens.Take(indexOfNextSentenceEnd).ToList();
+
+		        var indexOfToBeForm = sentence.FindIndex(token => toBeForms.Contains(token.Value));
+		        var indexOfPastParticiple = sentence.FindIndex(token => token.ExtendedType == "VBN");
+
+		        var isPassiveVoiceSentence = indexOfToBeForm != -1 && indexOfPastParticiple != -1 && indexOfToBeForm < indexOfPastParticiple;
+
+		        if (isPassiveVoiceSentence)
+		        {
+			        passiveVoiceSentencesCount++;
+		        }
+
+		        posTagTokens = posTagTokens.Skip(indexOfNextSentenceEnd + 1).ToList();
+                indexOfNextSentenceEnd = posTagTokens.FindIndex(token => token.ExtendedType == ".");
+            }
+
+	        return passiveVoiceSentencesCount / (double)totalSentencesCount;
+        }
```

Comment:
extract IsPassiveVoiceSentence so it would be smaller and more understandable

Reply:
alright.